src/engines/seq/euclid/euclid.cpp

```diff
@@ -204,7 +203,8 @@ namespace otto::engines {
     ctx.beginPath();
     ctx.fillText(
       util::join_strings(util::view::transform(
-                           util::view::filter(current.notes, [](char note) { return note >= 0; }),
+                           // extra safe on the note value to avoid overflow
+                           util::view::filter(current.notes, [](char note) { return note >= 0 && note < 128; }),
                            [](char note) { return midi::note_name(note); }),
```

Comment:
how can the note ever be greater than 127? its an unsigned byte...

Reply:
Hum, this is probably due to my poor comprehension of the "newest" C++ syntax, I was not sure how the declared type were applied. Now I think I get it better, indeed `< 128` is over the top. However in term of "extra safety" I considered adding a  `if (note >= 0) return midi::note_name(note); else return "MISS"; }` since it would not crash in case the filter iterator fails (e.g. nice for unit testing).

Reply:
note == -1 denotes a empty field in the array. I just use an array instead of a vector because its fine, and we avoid allocation.

Reply:
Yes, I got that :) Note that I am currently trying to resolve the remaining issues I mentioned in my branch, I will create a new pull request (since this one is now locked) when I finish.